R/add.R

```diff
@@ -137,16 +137,22 @@ add_column <- function(.data, ..., .before = NULL, .after = NULL) {
 
   pos <- pos_from_before_after_names(.before, .after, colnames(.data))
 
+  end_pos <- ncol(.data) + seq_len(ncol(df))
+
   if (pos <= 0L) {
-    out <- cbind(df, .data)
+    indexes_orig <- seq_len(ncol(.data))
+    .data[end_pos] <- df
+    .data <- .data[c(end_pos, indexes_orig)]
   } else if (pos >= ncol(.data)) {
-    out <- cbind(.data, df)
+    .data[end_pos] <- df
   } else {
-    indexes <- seq_len(pos)
-    out <- cbind(.data[indexes], df, .data[-indexes])
+    indexes_before <- seq_len(pos)
+    indexes_after <- (length(indexes_before)+1):ncol(.data)
```

Comment:
Can you please use `rlang::seq2()` instead of `:`? Perhaps this allows getting rid of one or even two branches in the `if`?

Reply:
Added `rlang::seq2()`.